apps/dc_tools/odc/apps/dc_tools/stac_api_to_dc.py

```diff
@@ -127,10 +127,8 @@ def stac_api_to_odc(
     **kwargs,
 ) -> Tuple[int, int]:
     # QA the BBOX
-    if config["bbox"]:
-        assert (
-            len(config["bbox"]) == 4
-        ), "Bounding box must be of the form lon-min,lat-min,lon-max,lat-max"
+    if config.get("bbox") and len(config["bbox"]) != 4:
```

Comment:
Rather than extracting "bbox" key twice, one with `.get` and then again with `[]` it's generally better practice to do it once, this also makes it thread-safe unlike current code (I know it does not matter in this code at this point in time)

```python
bbox = config.get("bbox", None)
if bbox is not None:
   if len(bbox) != 4:
       raise ValueError("Bounding box must be of the form lon-min,lat-min,lon-max,lat-max")
```
Also this will still fail if `bbox` is not a list, but some other construct, so should probably check for that. Or use some library that performs that kind of validation internally.

Reply:
hold on, is `config` our own construct or something coming from STAC API? or is it going to STAC API?

If it's our own then I would recommend using @dataclasses to explicitly define configuration set, then all the checking can happen there.